generators/app/templates/src/app/components/CustomText/index.js

```diff
@@ -0,0 +1,48 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Text } from 'react-native';
+
+import styles from './styles';
+
+export default class CustomText extends PureComponent {
+  static VARIANTS = [
+    'xxsmall',
+    'xsmall',
+    'small',
+    'medium',
+    'big',
+    'xbig',
+    'bold',
+    'center',
+    'white',
+    'blue',
+    'green',
+    'gray'
+  ];
+
+  customStyles = () =>
+    CustomText.VARIANTS
+      .map(variant => (this.props[variant] ? styles[variant] : null))
+      .filter(style => style !== null);
+
+  render() {
+    return (
+      <Text {...this.props.textProps} style={[styles.base, this.customStyles(), this.props.style]}>
+        {this.props.children}
+      </Text>
+    );
+  }
+}
+
+CustomText.defaultProps = {
+  textProps: {}
+};
+
+CustomText.propTypes = {
+  children: PropTypes.node.isRequired,
+  textProps: PropTypes.shape({ ...Text.propTypes })
```

Comment:
A value assigned as a propTypes cannot be a plain js object, it needs to be a propTypes instance.
That's why `textProps: Text.propTypes` will not work, and `textProps: PropTypes.shape({ ...Text.propTypes })` is needed